File: Round-8/傅立叶-勒让德展开/jury.c
#include <stdio.h>
#include <math.h>

double P[55][55];
int n, a, b;

double f(double x) {
    double ans = 0, tx = 1;
    for (int i = 0; i <= n; ++i) {
        ans += P[n][i] * tx;
        tx *= x;
    }
    ans *= cos(a * x + b);
    return ans;
}

double simpson(double l, double r) {
    double mid = l + (r - l) / 2;
    return (r - l) * (f(l) + 4 * f(mid) + f(r)) / 6;
}

double integral(double l, double r, int n) {
    double res = 0, h = (r - l) / n;
    for (int k = 0; k < n; ++k) {
        res += simpson(l + k * h, l + (k + 1) * h);
    }
    return res;
}

int main() {
    scanf("%d %d %d", &a, &b, &n);
    P[0][0] = 1;
    P[1][1] = 1;
    for (int i = 2; i <= n; ++i) {
        for (int j = 0; j <= i; ++j) {
            if (j) P[i][j] = P[i - 1][j - 1] * (2 * i - 1) / i;
            P[i][j] -= P[i - 2][j] * (i - 1) / i;
        }
    }
    printf("%lf\n", integral(-1, 1, 5000) * (2 * n + 1) / 2);
    return 0;
}

File: Round-8/傅立叶-勒让德展开/jury.py
from math import sin, cos

def legendre(n):
	ppp, pp, p = [1], [0, 1], None
	if n == 0:
		return ppp
	elif n == 1:
		return pp
	for i in range(2, n + 1):
		p = [0] * (len(pp) + 1)
		for j in range(len(pp)):
			p[j + 1] += (2 * i - 1) / i * pp[j]
		for j  in range(len(ppp)):
			p[j] -= (i - 1) / i * ppp[j]
		ppp, pp = pp, p
	return p

def fit(a, b, order):
	if a == 0:
		return cos(b) if order == 1 else 0
	p = legendre(order)
	r1, r2 = 0, 0
	ans = 0
	for i in range(len(p) - 1, -1, -1):
		f = r1 + p[i]
		n = (-1) ** i
		ans += f * (\
			1 / a * (sin(a + b) + n * sin(a - b)) + \
			i / a ** 2 * (cos(a + b) + n * cos(a - b))\
		)
		r1, r2 = r2, -f * i * (i - 1) / a ** 2
	return ans * (2 * order + 1) / 2

print(fit(*map(int, input().split(" "))))